### PR TITLE
[Core] Return early NodeTraverser::STOP_TRAVERSAL on referenced $isParamUsed is true on ParamAnalyzer

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
PHPStan notice is unrelated:

```bash
Run vendor/bin/phpstan analyse --ansi --error-format symplify
Note: Using configuration file /home/runner/work/rector-src/rector-src/phpstan.neon.
    0/1427 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
  720/1427 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░]  50%
 1427/1427 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
 ------------------------------------------------------------------------
  packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php:172
 ------------------------------------------------------------------------
  - '#Class method "getByType\(\)" is never used#'
 ------------------------------------------------------------------------
 ------------------------------------------------------------------------
  packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php:239
 ------------------------------------------------------------------------
  - '#Class method "getByAnnotationClass\(\)" is never used#'
 ------------------------------------------------------------------------


 [ERROR] Found 157 errors                                                      
```

ref https://github.com/rectorphp/rector-src/runs/7182500286?check_suite_focus=true

**Comment by samsonasik** [comment]
[ignored new phpstan notice](https://github.com/rectorphp/rector-src/pull/2630/commits/c32d3b7e76f1e9e05d4b5acaa5e29b735017636f)

File: phpstan.neon [hunk 1]
```diff
@@ -668,3 +668,5 @@ parameters:
             path: packages/NodeTypeResolver/PhpDocNodeVisitor/NameImportingPhpDocNodeVisitor.php
 
         - '#Class has a static method must so must contains "Static" in its name#'
+
+        - '#Class method ".+\(\)" is never used#'
```
Comment (TomasVotruba) [inline phpstan.neon:672]: Good solution :+1:  I'm currently testing the rule, so there will be some false positives.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Pull request merged**